src/main/java/com/hoverla/bring/context/CacheProxyConfigurator.java

```diff
@@ -0,0 +1,53 @@
+package com.hoverla.bring.context;
+
+import com.hoverla.bring.annotation.Cacheable;
+import net.sf.cglib.proxy.Enhancer;
+import net.sf.cglib.proxy.MethodInterceptor;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CacheProxyConfigurator implements ProxyConfigurator {
+
+    private Map<String, Map<List<Object>, Object>> cache = new ConcurrentHashMap<>();
+
+    @Override
+    public Object replaceWithProxyIfNeeded(Object t, Class<?> implClass) {
+        boolean isCacheable = Arrays.stream(implClass.getMethods()).anyMatch(m -> m.isAnnotationPresent(Cacheable.class));
+        if (isCacheable) {
+            MethodInterceptor methodInterceptor = (obj, method, args, proxy) -> {
+                if (method.isAnnotationPresent(Cacheable.class)) {
```

Comment:
Maybe collect the methods with `@Cacheable` annotation to the `Map` and then verify if the `Map` contains such a method instead of using Reflection one more time. The idea should be somewhat similar to the current homework, where we collect methods with `@Trimmed` annotation in `postProcessBeforeInitialization` and then reuse the `Map` in the `postProcessAfterInitialization` method.

Reply:
unfortunately I haven't done the homework yet, could we discuss it on quick call?

Reply:
yes, sure. I have some free time around 13:30